src/evict/evict_page.c

```diff
@@ -872,9 +874,24 @@ __evict_reconcile(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags)
          * outside world.
          */
         __wt_txn_bump_snapshot(session);
-    else if (use_snapshot_for_app_thread)
+    else if (use_snapshot_for_app_thread) {
+        /*
+         * If we couldn't make progress with the application thread's existing snapshot, save the
+         * existing snapshot and refresh to acquire a new one. Then try eviction again. Once the
+         * application threads are done with eviction, the application thread's snapshot is switched
+         * back to the original.
+         */
+        if (F_ISSET(session->txn, WT_TXN_REFRESH_SNAPSHOT)) {
+            F_CLR(session->txn, WT_TXN_REFRESH_SNAPSHOT);
```

Comment:
Can we move this(clear flag) line inside the `__wt_txn_snapshot_save_and_refresh` function?

Reply:
Thanks - the other comment (about not clearing the flag) made this redundant.